segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/cache/CacheLayout.java

```diff
@@ -0,0 +1,347 @@
+     * @return The length. This value is undefined if `blockMetadata` was not generated using one of the methods in this class
+     * or if {@link #isUsedBlock} returns false on `blockMetadata`.
+     */
+    abstract int getLength(long blockMetadata);
+
+    /**
+     * Updates the given block metadata for an unallocated Buffer-Block to point to the next next free (unallocated) Block-Buffer.
+     * @param blockMetadata The block metadata to update.
+     * @param nextFreeBlockId The Buffer-Block id of the next free block.
+     * @return The resulting block metadata. This value is undefined if `blockMetadata` was not generated using one of
+     * the methods in this class.
+     */
+    abstract long setNextFreeBlockId(long blockMetadata, int nextFreeBlockId);
+
+    /**
+     * Gets the next free block id from the given block metadata.
+     * @param blockMetadata The block metadata to get from.
+     * @return The next free block id. This value is undefined if `blockMetadata` was not generated using one of the methods
+     * in this class or if {@link #isUsedBlock} returns true on `blockMetadata`.
+     */
+    abstract int getNextFreeBlockId(long blockMetadata);
+
+    /**
+     * Gets a value indicating whether the Buffer-Block associated with the given block metadata is used or not.
+     * @param blockMetadata The block metadata to query.
+     * @return True if used, false otherwise. This value is undefined if `blockMetadata` was not generated using one of
+     * the methods in this class.
+     */
+    abstract boolean isUsedBlock(long blockMetadata);
+
+    /**
+     * Generates a new Buffer-Block Metadata having IsUsed set to true.
+     * @param nextFreeBlockId    The Id of the next unallocated Buffer-Block.
+     * @param length             The length of the data in this Buffer-Block.
+     * @param predecessorAddress The address of the next block in the sequence.
+     * @return A new block metadata.
+     */
+    abstract long newBlockMetadata(int nextFreeBlockId, int length, int predecessorAddress);
+
+    /**
+     * Generates a new Buffer-Block metadata with no contents.
+     * @return A new block metadata.
+     */
+    abstract long emptyBlockMetadata();
+
+    /**
+     * Converts the given address into a readable format (used for logging).
+     * @param address The address to decode.
+     * @return A String. This value is undefined if `address` was not generated using one of the methods in this class.
+     */
+    String getAddressString(int address) {
+        return address == NO_ADDRESS ? "" : String.format("Buffer = %d, Block = %d", getBufferId(address), getBlockId(address));
+    }
+
+    //endregion
+
+    //region Default Layout
+
+    /**
+     * {@link CacheLayout} with the following characteristics:
+     * - 2MB Buffers
+     * - 4KB block size
+     * - 511 usable Buffer-Blocks per Buffer.
+     *
+     * Address layout (26 bits used out of 32)
```

Comment:
It seems like somewhere the fact that this allows a max size of 256gb should be defined as a constant.

Reply:
This is: `CacheLayout.MAX_TOTAL_SIZE`.  I have just updated it to be 256GB (it had an smaller value, which was from a previous choice of layout).